// fancylog/lib/fileLogger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var strip_ansi_1 = __importDefault(require("strip-ansi"));
var construct_1 = require("./construct");
var errors_1 = require("./errors");
var fs_1 = __importDefault(require("./fs"));
var FileLogger = (function () {
    function FileLogger(path) {
        if (path === undefined || path === null || path === '') {
            throw new Error(errors_1.INVALID_PATH);
        }
        this.path = path;
    }
    FileLogger.prototype.info = function (object) {
        this.log('info', object);
    };
    FileLogger.prototype.debug = function (object) {
        this.log('debug', object);
    };
    FileLogger.prototype.error = function (object) {
        this.log('error', object);
    };
    FileLogger.prototype.verbose = function (object) {
        this.log('verbose', object);
    };
    FileLogger.prototype.warn = function (object) {
        this.log('warn', object);
    };
    FileLogger.prototype.log = function (logLevel, object) {
        var _this = this;
        var str = construct_1.construct(logLevel, object);
        var stripped = strip_ansi_1.default(str) + "\n";
        console.log(str);
        fs_1.default
            .ensureFile(this.path)
            .then(function () { return fs_1.default.appendFile(_this.path, stripped); })
            .catch(function (err) {
            throw err;
        });
    };
    return FileLogger;
}());
exports.FileLogger = FileLogger;
